api/v2/Routes/isadmin/index.js

```diff
@@ -7,13 +7,25 @@ const router = Router({ mergeParams: true });
 
 // Responds with whether or not the current user is an admin
 router.get('/', async (req, res) => {
-    const authHeader = req.headers['authorization'];
-    if (!authHeader) {
-        throw new AuthorizationError("Authorization Header is empty.");
+    let adminStatus = false;
+    try {
+        const authHeader = req.headers['authorization'];
+        if (!authHeader) {
+            throw new AuthorizationError("Authorization Header is empty.");
+        }
+        const authEmail = await getEmailFromAuth(authHeader);
+        adminStatus = await isAdmin(authEmail);
```

Comment:
```suggestion
        const adminStatus = await isAdmin(authEmail);
```

Reply:
Totally as an aside for now or later, but we can probably parallelize these with `promise.all`.  We should also return a 403 if the user isn't admin because they aren't going to magically get admin any time soon haha.  See the difference between 401 and 403 on MDN